### ISSUE TITLE
Icons stuck to Personal Resource Display

### ISSUE THREAD

**Issue by Vizcro** [jordonwow/bigdebuffs#234]
Hi,

please see the attached picture for the issue.

![BigDebuffs_issue](https://user-images.githubusercontent.com/79589653/109020755-89a1a980-76ba-11eb-9c66-d618677b4f2d.jpg)

It always seems to be an icon that previously appeared on my target's nameplate (e.g. for my Mage it's almost exlusively CS/Frost Nova in PvE). The icon is stuck there until /reload. It's really distracting and annoying :(
Unfortunately I haven't been able to pinpoint what exactly causes this, but it happens quite frequently: pretty much during every battleground and like every 2nd or 3rd arena match.

**Comment by domzae** [comment]
Did this only start happening after updating to 9.9 today?

**Comment by Vizcro** [comment]
No, this has been happening since I started using the addon (like 2 weeks ago).

**Comment by Kennah229** [comment]
> Did this only start happening after updating to 9.9 today?

You haven't seen this issue before? Maybe you don't use Personal Resource Display. I've had this issue many many times since first installing the addon like 6 months ago. I've tried to figure out how to fix it but couldn't figure it out.

**Comment by Vizcro** [comment]
Is there hope for this to get fixed anytime soon? I'm ofc ready to help pinning down the issue if needed.
I'm very grateful this addon exists and I'm aware the author is putting a lot of hard work into it already, but this is an almost dealbreaking issue for me (and everyone I know of that uses the Personal Resource Display), as it is extremely distracting.

**Comment by domzae** [comment]
I've never used the personal resource display - I'm guessing it is treated as a nameplate - does disabling BigDebuffs from showing on nameplates stop this from happening?

**Comment by Vizcro** [comment]
> I've never used the personal resource display - I'm guessing it is treated as a nameplate - does disabling BigDebuffs from showing on nameplates stop this from happening?

Indeed it does.

**Comment by domzae** [comment]
I played a little with the PRD enabled, but never saw this bug :(. 

You can try going to Line 1919 in BigDebuffs.lua, and change the line from
`if namePlate:IsForbidden() then return end`
to
`if namePlate:IsForbidden() or UnitIsUnit("player", unit)" then return end`

**Comment by Vizcro** [comment]
> I played a little with the PRD enabled, but never saw this bug :(.
> 
> You can try going to Line 1919 in BigDebuffs.lua, and change the line from
> `if namePlate:IsForbidden() then return end`
> to
> `if namePlate:IsForbidden() or UnitIsUnit("player", unit)" then return end`

Thank you for trying. Have you joined a battleground? That‘s where it most commonly happens. I‘ll try your fix today. If it doesn‘t work I‘ll try to find out how to reproduce the bug.

**Comment by LewisFarrellDev** [comment]
I have also experienced this alongside icons staying on Friendly Nameplates even though that option is disabled. It seems like a bigger problem with Icons either not upating/refreshing themselves for whatever reason. The problem is highly noticable after PvP combat of any kind. 

See my seperate logged issue for reference #236

**Comment by Vizcro** [comment]
You can see the bug appearing in [this video](https://youtu.be/tdcDdGxtr7A?t=530) from Supatease at around 8:50. The enemy Feral uses Tiger's Fury and all of a sudden it's stuck to the PRD and remains there even after the duration has ended. Interestingly enough the Feral wasn't even the target or focus target (up to this point I assumed it would only mirror buffs from the target or focus target). The moment the bug appears also seems to line up perfectly with the player switching forms. It looks to me like the Feral popping TF and the player switching forms happened at the same time, so it might also be related to the PRD "refreshing" itself the moment he's switching forms.

**Comment by domzae** [comment]
@Vizcro Okay, I have another fix you can try:

Delete (or comment with `--`)  Line 1791  
```
or (UnitIsUnit("player", unit))
```

Add after Line 1800 
```
    if UnitIsUnit("player", unit) then 
        frame:Hide()
        frame.current = nil
        return
    end
```

**Comment by LewisFarrellDev** [comment]
Fix didn't work for me :(

**Comment by domzae** [comment]
Damn, another potential work around would be to only delete line 1791 above - that should mean buffs/debuffs will display on your nameplate, which shouldat least clear any bugged buffs

**Comment by LewisFarrellDev** [comment]
Would this maybe Address this issue though to? #236 

Seems like there is a general issue with Icons sticking on any friendly nameplates

**Comment by Vizcro** [comment]
I didn't get to try your fixes yet, will let you know once I get the chance this weekend. Thanks again for trying to figure this out!

**Comment by Vizcro** [comment]
> Damn, another potential work around would be to only delete line 1791 above - that should mean buffs/debuffs will display on your ~nameplate~ Personal Resource Display, which should at least clear any bugged buffs

Unfortunately the fixes don't work, but the workaround does. It seems that nothing gets stuck on the PRD that way. Once the current tracked buff runs out, it just disappears (or so it seems, but I haven't done too much testing with it yet). Ofc this doesn't fix the issue of a fat distracting icon on the PRD, but it helps somewhat.
This might be dumb idea, but to add to that workaround, would it be possible to simply make the icons on the PRD clickthrough and decrease their alpha to zero?

**Comment by domzae** [comment]
Perhaps an easier way to do what you're suggesting, is to "delete" the icon right after we generate it.  (Using the line numbers from v9.13)

Remove line 1798:
~~```--or (UnitIsUnit("player", unit))```~~

Add at 1909:
```    
if (UnitIsUnit("player", unit)) and frame.current ~= nil then 
    frame:Hide()
    frame.current = nil
end
```
Again though, it won't actually fix the bug, but it should clear it whenever you get an aura that would otherwise show on your 'nameplate'.

**Comment by Vizcro** [comment]
> Again though, it won't actually fix the bug, but it should clear it whenever you get an aura that would otherwise show on your 'nameplate'.

It worked! No icons appearing on the PRD anymore, as far as I can tell after some testing in PvP and PvE.

Removing line 1798 (previously 1791) already helped in the sense that I had no bugged icons appear anymore, like none at all. Adding the lines at 1909 seems to have fixed it entirely for me. I'll have to do some more testing, but I'm fairly certain you did it, @domzae.

**Comment by Vizcro** [comment]
After some additional testing I can confirm this fix works. I know it's not perfect and the underlying problem technically still kinda (?) persists, but it's a functional bandaid. As such, this should imo be part of the next addon update so everyone is finally free of random icons stuck to the PRD @jordonwow @domzae.

**Comment by domzae** [comment]
Glad to hear! I wonder if the same kind of 'band-aid' fix will also work for #236. I've made the change in my fork. ([https://github.com/domzae/bigdebuffs/commit/fixstuckicons](https://github.com/domzae/bigdebuffs/commit/3cfc0d944d4fe0cd5d03cc14532ba06efd6a0e73))
If this fixes icons getting stuck on other nameplates too let me know and I'll open a pull request! :)
